firmware/src/touch.c

```diff
@@ -155,16 +155,15 @@ bool touch_sensor_ok(unsigned i)
 
 const uint16_t *touch_raw()
 {
-    static uint16_t readout[36];
-    uint16_t buf[36];
+    static uint16_t readout[36] = {0};
+    // Do not use readout as buffer directly, update readout with buffer when operation finishes
+    uint16_t buf[36] = {0};
 
     for (int i = 0; i < 3; i++) {
         sensor_ok[i] = mpr121_raw(MPR121_BASE_ADDR + i, buf + i * 12, 12);
     }
+    memcpy(readout, buf, sizeof(readout));
 
```

Comment:
Just revert to my original codes.
Let this PR purely focus on "better raw command display".

Reply:
I think it's quite essential to modify the `touch_raw()` function because it originally returns data realigned to zones, it will be frustrating to map it back to data aligned by sensors. It may be better to map to zones in `handle_raw()` or another new function like `map_to_zones()`.

This function is only referenced at `commands.c/handle_raw()`, thus this modification won't affect other functionalities.

Reply:
> I think it's quite essential to modify the `touch_raw()` function because it originally returns data realigned to zones, it will be frustrating to map it back to data aligned by sensors. It may be better to map to zones in `handle_raw()` or another new function like `map_to_zones()`.
> 
> This function is only referenced at `commands.c/handle_raw()`, thus this modification won't affect other functionalities.

I see your point. Go ahead.